### PR TITLE
Added clean up to scheduler-predicates

### PR DESCRIPTION
**Pull request by piosz**

### CODE REVIEW

**Comment by piosz** [comment]
cc @smarterclayton @roberthbailey @wojtek-t 

This change should help with failing Scheduler Predicates e2e test in addition to #12901

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -228,6 +238,7 @@ var _ = Describe("SchedulerPredicates", func() {
 		time.Sleep(10 * time.Second)
 
 		verifyResult(c, podName, ns, currentlyDeadPods)
+		cleanupPods(c, ns)
```
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: Maybe put it at the end of verifyResult?

**Comment by gmarek** [comment]
One nit, LGTM otherwise.

File: test/e2e/scheduler_predicates.go (hunk 1 above)
Comment (piosz) [inline test/e2e/scheduler_predicates.go]: Is it a part of verifying results?

Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: Yeah... you're probably right.

**Pull request merged**